mountpoint-s3/tests/fs.rs

```diff
@@ -1232,3 +1232,65 @@ async fn test_flexible_retrieval_objects() {
         }
     }
 }
+
+#[tokio::test]
+async fn test_readdir_rewind() {
+    let (client, fs) = make_test_filesystem("test_readdir_rewind", &Default::default(), Default::default());
+
+    for i in 0..10 {
+        client.add_object(&format!("foo{i}"), b"foo".into());
+    }
+
+    let dir_handle = fs.opendir(FUSE_ROOT_INODE, 0).await.unwrap().fh;
+
+    let mut reply = DirectoryReply::new(5);
+    let _ = fs
+        .readdirplus(FUSE_ROOT_INODE, dir_handle, 0, &mut reply)
+        .await
+        .unwrap();
+    let entries = reply
+        .entries
+        .iter()
+        .map(|e| (e.ino, e.name.clone()))
+        .collect::<Vec<_>>();
+    assert_eq!(entries.len(), 5);
+
+    fs.readdirplus(FUSE_ROOT_INODE, dir_handle, 1, &mut Default::default())
+        .await
+        .expect_err("out of order");
+
+    let mut new_reply = DirectoryReply::new(5);
+    let _ = fs
+        .readdirplus(FUSE_ROOT_INODE, dir_handle, 0, &mut new_reply)
+        .await
+        .unwrap();
+    let new_entries = new_reply
+        .entries
+        .iter()
+        .map(|e| (e.ino, e.name.clone()))
+        .collect::<Vec<_>>();
```

Comment:
Related to the other comment, I'm wondering what the behaviour should be if this was `DirectoryReply::new(3)` for the repeated `readdirplus`. I'm imagining out of order, and then we can still repeat with the same offset and reply size and it would be OK?

Reply:
I added more test cases to make this clear -- the size of the buffer never matters, only the offset.